src/core/server/integration_tests/saved_objects/migrations/group4/v2_md5_to_mv.test.ts

```diff
@@ -0,0 +1,234 @@
+      expect(await getMappingMeta()).toEqual({
+        indexTypesMap: {
+          '.kibana': ['another-type', 'some-type'],
+        },
+        migrationMappingPropertyHashes: {
+          'another-type': '10.1.0',
+          'some-type': '10.1.0',
+        },
+      });
+    });
+
+    describe('when upgrading to a more recent version', () => {
+      beforeAll(async () => {
+        // start the migrator again, which will update meta with modelVersions
+        const { runMigrations: restartKibana } = await getKibanaMigratorTestKit({
+          kibanaIndex: MAIN_SAVED_OBJECT_INDEX,
+          types: [SOME_TYPE, ANOTHER_TYPE_UPDATED], // note that we update 'another-type'
+          hashToVersionMap: HASH_TO_VERSION_MAP,
+          logFilePath,
+        });
+
+        result = await restartKibana();
+      });
+      it('updates the SO indices meta.migrationMappingPropertyHashes with the appropriate model versions', async () => {
+        expect(result[0].status).toEqual('patched'); // should be a compatible (non-reindexing) migration
+
+        expect(await getMappingMeta()).toEqual({
+          indexTypesMap: {
+            '.kibana': ['another-type', 'some-type'],
+          },
+          migrationMappingPropertyHashes: {
+            'another-type': '10.2.0', // switched to modelVersion: 2
+            'some-type': '10.1.0',
+          },
+        });
+      });
+
+      it('only "picks up" the types that have changed', async () => {
+        const logs = await readLog(logFilePath);
+        expect(logs).toMatch(
+          'Kibana is performing a compatible upgrade and NO root fields have been udpated. Kibana will update the following SO types so that ES can pickup the updated mappings: another-type.'
+        );
+      });
+    });
+  });
+
+  describe('when SO indices still contain md5 hashes', () => {
+    beforeAll(async () => {
+      const { runMigrations, client } = await getKibanaMigratorTestKit({
+        kibanaIndex: MAIN_SAVED_OBJECT_INDEX,
+        types: [SOME_TYPE, ANOTHER_TYPE],
+        logFilePath,
+      });
+      esClient = client;
+
+      // misc cleanup
+      await clearLog(logFilePath);
+      await deleteSavedObjectIndices(client);
+
+      await runMigrations();
+
+      // we update the mappings to mimic an "md5 state"
+      await client.indices.putMapping({
+        index: MAIN_SAVED_OBJECT_INDEX,
+        _meta: {
+          migrationMappingPropertyHashes: {
+            'some-type': SOME_TYPE_HASH,
+            'another-type': ANOTHER_TYPE_HASH,
+          },
+        },
+        allow_no_indices: true,
+      });
+
+      // we then start the migrator again, which will update meta with modelVersions
+      const { runMigrations: restartKibana } = await getKibanaMigratorTestKit({
+        kibanaIndex: MAIN_SAVED_OBJECT_INDEX,
+        types: [SOME_TYPE, ANOTHER_TYPE_UPDATED],
+        hashToVersionMap: HASH_TO_VERSION_MAP,
+        logFilePath,
+      });
+
+      result = await restartKibana();
```

Comment:
Looks like we are missing some asserts here

Reply:
👍🏼 I refactored a bit these tests, and added asserts for the expected `result`.